### PR TITLE
feature:add rabbit retries

### PR DESCRIPTION
**Pull request by moranshemesh**

### CODE REVIEW

File: lib/sender.js [hunk 1]
```diff
@@ -2,17 +2,20 @@
...
+var DEFAULT_MESSAGE_TIMEOUT_INTERVAL = 5000;
 var _ = require('lodash');
 
 /**
  * Responsible for sending messages to the AMQP server
  * @param channel
  * @constructor
  */
-function Sender(channel, errorExchange, nackExchange) {
+function Sender(channel, errorExchange, nackExchange, timeoutExchange, timeoutQueueName) {
```
Comment (asafdav) [inline lib/sender.js]: I would rename to timeoutQueueNamePrefix

File: lib/sender.js [hunk 2]
```diff
@@ -79,16 +82,42 @@ Sender.prototype.error = function(msg, reason) {
...
   attempts = attempts ? attempts - 1 : DEFAULT_MAX_ATTEMPTS - 1;
+
+  var timeoutInterval = msg.properties.headers['time-out-interval'] ?
+    msg.properties.headers['time-out-interval'] : DEFAULT_MESSAGE_TIMEOUT_INTERVAL;
   if (attempts > 0) {
-    var options = buildOptionsFromMsg(msg, {'x-attempts-left': attempts});
-    this.publish(this.nackExchange, msg.fields.routingKey, msg.content, options);
+    var options = this.buildOptionsFromMsg(msg, {
+      'x-attempts-left': attempts,
+      'message-ttl': timeoutInterval
```
Comment (asafdav) [inline lib/sender.js]: I would add a comment that describes why we need both a queue and per message ttl

File: lib/sender.js [hunk 3]
```diff
@@ -79,16 +82,42 @@ Sender.prototype.error = function(msg, reason) {
...
+    });
+    //make sure we have the correct queue and bind it to the EXCHANGE
+    var queueName = this.timeoutQueueName + '.timeout.' + timeoutInterval;
+
+    var self = this;
+    this.channel.assertQueue(queueName, {
+        deadLetterExchange: self.nackExchange,
+        arguments: {
+          'x-message-ttl': timeoutInterval,
+          'x-expires': 90000
```
Comment (asafdav) [inline lib/sender.js]: could be a const or a variable as well

File: lib/sender.js (hunk 2 above)
Comment (asafdav) [inline lib/sender.js]: you also need it for x-match

Comment (moranshemesh) [inline lib/sender.js]: done.

**Pull request merged**